src/server/handlers/launchpad.js

```diff
@@ -387,10 +387,17 @@ const internalFindSnaps = async (owner, token) => {
       owner: `/~${username}`
     }
   });
+
+  // The response may be split across multiple batches.  Gather them all.
+  const entries = [];
+  for await (const entry of result) {
+    entries.push(entry);
+  }
```

Comment:
The batching exists for a reason, and the same reason applies to build.snapcraft.io.  This is an unboundedly large set, and could easily result in timeouts.

Instead, we need to have corresponding batching for this build.snapcraft.io endpoint.  But in order to do that in a stable way, I think we need to be able to do cursor-based navigation of Launchpad's snap build batches, and Launchpad doesn't support that yet.  That's why I hadn't already done this.  I made a start on the Launchpad side last September, but my branch still needs more work.

Reply:
Oh actually, this is just snaps for a prefix isn't it, not snap builds?  That's probably OK then, as while it may well go over a single batch it won't be arbitrarily long in practice.  Carry on.